docs/5.x upgrade guide.md

```diff
@@ -231,6 +231,7 @@ render((
 In version 5, all the utility functions that were previously accessed via `import { utils } from '@rjsf/core';` are now available via `import utils from '@rjsf/utils';`.
 Because of the decoupling of validation from `@rjsf/core` there is a breaking change for all the [validator-based utility functions](https://react-jsonschema-form.readthedocs.io/en/stable/api-reference/utiltity-functions#validator-based-utility-functions), since they now require an additional `ValidatorType` parameter.
 More over, one previously exported function `resolveSchema()` is no longer exposed in the `@rjsf/utils`, so use `retrieveSchema()` instead.
+Finally, one previously exported function `getMatchingOption()` has been deprecated in favor of `getFirstMatchingOption()`.
```

Comment:
What's the reason for renaming it to `getFirstMatchingOption`? I think `getMatchingOption` is pretty clear ("option" not "options") and not sure if "first" is worth the rename?

Reply:
the `getMatchingOption()` function logic loops over all of the options in a list and return the first option that is considered valid for a schema. This means that it stops looking for any other options. Sometimes there is actually an option that has a closer match to the formData that the first one. Which is why I added the `getClosestMatchingOption()` function. The rename better suggests the behaviour

Reply:
i have a _big_ problem right now where i have `anyOf` where two schemas have an intersecting key. if my form is empty _except_ that common key, sometimes, _the form just changes views_, which is extremely disruptive.

that is:

```json
{ type: "object", properties: { "foo": ..., ...A } }
{ type: "object", properties: { "foo": ..., ...B } }
```

...the form can toggle. hopefully this changeset will make it clear that if a current alternative is selected, it _should not change_ the active view

Reply:
@cdaringe I plan on cutting beta-18 with this change within a day. Let me know if does not fix it, but from what you shared I believe it would work properly. What is this key's type? (i.e. what is `A` and `B` defined to be?

Reply:
hey @heath-freenome , thanks! i'll install the beta as soon as it's out and test if it's fixed. A & B are objects of _other_ properties that are non-insersecting. in the above, i was poorly trying to communicate that that I have `anyOf` with 2 schemas, which share the field `foo` but share no other field names.